lib/model/QubitRepository.php

```diff
@@ -192,11 +213,16 @@ public function delete($connection = null)
     // Remove adv. search repository options from cache
     QubitCache::getInstance()->removePattern('search:list-of-repositories:*');
 
-    foreach ($this->informationObjects as $item)
+    foreach ($this->getRelatedInformationObjectIds() as $id)
     {
+      $item = QubitInformationObject::getById($id);
+
       unset($item->repository);
 
       $item->save();
+
+      // Keep caches clear to prevent memory use from ballooning
+      Qubit::clearClassCaches();
```

Comment:
@mcantelon, it's great that you spotted the same issue on the deletion, thanks!!

While this changes will prevent the memory problem, the process is still prone to timeouts. Also, it looks like this will trigger a job for each related IO with descendants, when it could be done in a single one (like on the save method).

I understand there are time constrains for this fix and the repository deletion (specially with a lot of related IOs) is not as normal as its modification, so I'll create a related ticket for this particular matter.

Reply:
Actually, we could use the same one so I've added a note and modified the title of https://projects.artefactual.com/issues/13250.

Reply:
Sounds good!